### PR TITLE
QPP Analysis

### PR DESCRIPTION
**Pull request by hscells**
This commit has a lot of minor bug fixes from when I was trying to do QPP analysis.

Namely, it:

 - Adds headers to the QPP output file
 - Fixes a couple of bugs with QPPs
 - Move the KLDivergence class into the LanguageModel class (this might be the wrong thing to have done)
 - Add a class for getting all the terms in an index
 - Add a script for extracting citations out of the pubmed files by pmid (in parallel) 
 - Actually store the term vector information if specified when indexing PubMed

### CODE REVIEW

File: scripts/pubmedextract.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    decompressed_file = gzip.GzipFile(file, mode='rb')
+    tree = etree.parse(decompressed_file)
+    root = tree.getroot()
+    for node in root.findall('PubmedArticle'):
+        pmid = node.find('MedlineCitation').find('PMID').text
+        if pmid in pmids:
+            print(pmid)
+            file_data = data.format(
+                etree.tostring(node, encoding='unicode', method='xml', pretty_print=True))
+            with open('/datadrive2/pubmed_filter/{}.xml'.format(pmid), 'w') as f:
```
Comment (diegoceccarelli) [inline scripts/pubmedextract.py]: absolute path?

Comment (hscells) [reply scripts/pubmedextract.py]: Nice spotting!

Comment (leifos) [reply scripts/pubmedextract.py]: thanks

Comment (diegoceccarelli) [reply scripts/pubmedextract.py]: @leifos lol maybe wait for @hscells to fix the path before merging 💃

**Pull request merged**